### PR TITLE
Simplify resources calculation v2

### PR DESCRIPTION
**Pull request by piotmag769**
The same changes as in #1669 but concerning `deploy` and `deploy_at`

### CODE REVIEW

File: crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/deploy.rs [hunk 1]
```diff
@@ -28,30 +23,20 @@ use starknet_api::transaction::Calldata;
 use super::CheatcodeError;
 
 pub fn deploy_at(
-    blockifier_state: &mut BlockifierState,
+    syscall_handler: &mut SyscallHintProcessor,
```
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/deploy.rs:26]: This is the business logic change

File: crates/cheatnet/tests/common/mod.rs [hunk 2]
```diff
@@ -65,7 +68,113 @@ pub fn deploy_contract(
...
+    deploy(
+        &mut syscall_hint_processor,
+        runtime_state,
+        &class_hash,
+        calldata,
+    )
+    .unwrap()
+}
+
+pub fn deploy_wrapper(
```
Comment (piotmag769) [inline crates/cheatnet/tests/common/mod.rs:104]: Added wrapper functions for tests so syscall handlers are constructed automatically (they are not used anywhere in cheatnet tests + cheatnet tests need massive refactors anyways)

**Review by piotmag769 — COMMENTED** [review]
Only those files that are commented are relevant, the rest are signature changes

File: crates/cheatnet/tests/common/mod.rs [hunk 3]
```diff
@@ -65,7 +68,113 @@ pub fn deploy_contract(
...
+        &mut execution_resources,
+        &mut entry_point_execution_context,
+        Relocatable {
+            segment_index: 0,
+            offset: 0,
+        },
+        CallEntryPoint::default(),
+        &hints,
+        ReadOnlySegments::default(),
+    );
```
Comment (drknzz) [inline crates/cheatnet/tests/common/mod.rs:93]: Could we extract to function?

**Review by drknzz — APPROVED** [review]

File: crates/cheatnet/tests/common/mod.rs (hunk 3 above)
Comment (piotmag769) [reply crates/cheatnet/tests/common/mod.rs:93]: Yea, it is a pain to do, will do the most dumb macro ever I guess xD

File: crates/sncast/src/starknet_commands/script.rs [hunk 4]
```diff
@@ -300,15 +297,7 @@ pub fn run(
...
-    let block_context = build_default_block_context();
-    let account_context = build_transaction_context();
-    let mut context = EntryPointExecutionContext::new(
-        &block_context.clone(),
-        &account_context,
-        ExecutionMode::Execute,
-        false,
-    )
-    .unwrap();
+    let mut context = build_context(BlockInfo::default());
```
Comment (piotmag769) [inline crates/sncast/src/starknet_commands/script.rs:300]: Did it bcs I moved `build_context` function to `runtime` package, so we don't repeat the code

File: crates/cheatnet/tests/common/mod.rs (hunk 3 above)
Comment (piotmag769) [reply crates/cheatnet/tests/common/mod.rs]: Done

**Review by Arcticae — APPROVED** [review]

**Pull request merged**